radix-engine/src/engine/id_allocator.rs

```diff
@@ -1,99 +1,98 @@
 use scrypto::types::*;
 use scrypto::utils::*;
 
-pub const ECDSA_TOKEN_RID: Rid = Rid(0);
 pub const ECDSA_TOKEN_BID: Bid = Bid(0);
-const USER_RID_START: u32 = 1024;
-const USER_BID_START: u32 = 1024;
+pub const ECDSA_TOKEN_RID: Rid = Rid(1);
+
+pub const SYSTEM_OBJECT_ID_RANGE: (u32, u32) = (0, 512);
+pub const TRANSACTION_OBJECT_ID_RANGE: (u32, u32) = (512, 1024);
+pub const USER_OBJECT_ID_RANGE: (u32, u32) = (1024, u32::MAX);
```

Comment:
Seems like this would be better implemented as an enum as there should really never be an instance where an IdAllocator is used with user defined range.

Reply:
Good point. Will apply